src/core/engine/processors/kitchen_simulator.py

```diff
@@ -9,5 +12,49 @@ def __init__(self, app_processor_config, app_context=None):
             app_processor_config=app_processor_config, app_context=app_context
         )
 
+        ioc = get_ioc_instance()
+        self.order_controller = ioc.get_instance("order_controller")
+        self.order_status_history_controller = ioc.get_instance(
+            "order_status_history_controller"
+        )
+        self.chef_controller = ioc.get_instance("chef_controller")
+        self.order_manager = OrderManager()
+
     def process(self, delta_time):
         print("Waiting orders")
+        self.assign_orders_to_available_chefs()
+
+    def assign_orders_to_available_chefs(self):
+
+        orders_to_process = self.order_controller.get_orders_to_process()
+        orders_validation_map = self.order_controller.get_validated_orders_map(
+            orders_to_process
+        )
+        available_chef_ids = self.chef_controller.get_available_chefs()
+
+        if available_chef_ids:
+
+            order_in_turn_id = self.order_manager.get_queue_from_status("new_order")
+            order_to_be_assigned = list(
+                filter((lambda order: order.id == order_in_turn_id), orders_to_process)
+            )
+
+            if orders_validation_map[order_to_be_assigned[0].id]:
+                order_to_be_assigned[0].assigned_chef_id = available_chef_ids[0]
```

Comment:
I feel you can create a function the if/else processes like `assign_to_chef_and_send_to_in_process` and the other one `*_send_to_cancel`

Reply:
done